### PR TITLE
Refactor code to combine parseMedicalHistory and parseMedical as both…

### LINKED ISSUE #79: Refactor Code
Refactor the medicalHistory duplicate functions in parserUtil

### PR DESCRIPTION
**Pull request by kanna-1**
… share the same functionality

Resolves #79

### CODE REVIEW

**Comment by WZWren** [comment]
This refactoring is a good change! LGTM.

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 1]
```diff
@@ -203,18 +203,18 @@ public void parseEmail_validValueWithWhitespace_returnsTrimmedEmail() throws Exc
     }
     @Test
     public void parseMedicalHistory_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> ParserUtil.parseMedicalHistories(null));
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseMedicals(null));
```
Comment (WZWren) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:206]: It's good that you adapted the test cases for `parseMedicals`, but I think `parseMedicals` already has its own test cases from `Line 313` onwards. Have you checked if there are overlaps in test cases?

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 2]
```diff
@@ -223,18 +223,18 @@ public void parseMedicalHistory_collectionWithValidTags_returnsMedicalHistorySet
...
     public void parseMedicalHistory_invalidValue_throwsParseException() {
-        assertThrows(ParseException.class, () -> ParserUtil.parseMedicalHistory(INVALID_MEDICAL_HISTORY));
+        assertThrows(ParseException.class, () -> ParserUtil.parseMedical(INVALID_MEDICAL_HISTORY));
     }
 
 
     @Test
     public void parseTag_emptyValue_throwsParseException() {
-        assertThrows(NullPointerException.class, () -> ParserUtil.parseMedicalHistories(null));
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseMedicals(null));
```
Comment (WZWren) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:237]: If this is a test case that we are keeping, the name of the test case should be `parseMedical` or `parseMedicalHistory`.

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 3]
```diff
@@ -266,7 +266,7 @@ public void parseMedicalHistories_emptyHistory_exceptionThrown() {
         // Input data with an empty medical history
         String input = "History 1, , History 3";
 
-        assertThrows(ParseException.class, () -> ParserUtil.parseMedicalHistories(Arrays.asList(input.split(","))));
+        assertThrows(ParseException.class, () -> ParserUtil.parseMedicals(Arrays.asList(input.split(","))));
```
Comment (WZWren) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:269]: See comment at `Line 206` above.

**Review by WZWren — COMMENTED** [review]
This is a good change to reduce tech debt for us! If we can confirm if the test cases are redundant, this would be good to merge.

**Comment by kanna-1** [comment]
The new commit addresses the problem of redundant test cases.

**Comment by WZWren** [comment]
LGTM!

**Pull request merged**